### DISCUSSION TITLE
Case management use case

### DISCUSSION THREAD

**Discussion by mcrodriguezb** [serverlessworkflow/specification#793] [Q&A]
Hi all,

I'm trying to implement a case management scenario using the SW specification. It is clear I will have states, but not sure if the type would be event (with multiple events) or operations. 

In each state, I might have different actions to execute, mandatory or optionals. Mandatory actions could be implemented with events (exclusive=false), but not sure what to do with the optionals.

How would you implement such scenario?

**Comment by Maslino** [comment]
Switch state may be used to decide to execute optional actions or not.

**Comment by cdavernas** [comment]
Closing the discussion as it applies to a discontinued version of the spec.

@mcrodriguezb I invite you to look at our brand new [listen](https://github.com/serverlessworkflow/specification/blob/main/dsl-reference.md#listen) and [switch](https://github.com/serverlessworkflow/specification/blob/main/dsl-reference.md#listen) tasks, in case it helps!
